conans/client/generators/msbuild.py

```diff
@@ -0,0 +1,208 @@
+import os
+import textwrap
+from xml.dom import minidom
+
+from conans.client.tools import msvs_toolset, VALID_LIB_EXTENSIONS
+from conans.errors import ConanException
+from conans.model import Generator
+from conans.util.files import load
+
+
+class MSBuildGenerator(Generator):
+
+    dep_props = textwrap.dedent("""\
+        <?xml version="1.0" encoding="utf-8"?>
+        <Project ToolsVersion="4.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+            <ImportGroup Label="PropertySheets" >
+            </ImportGroup>
+            <PropertyGroup Label="UserMacros" />
+            <PropertyGroup />
+            <ItemDefinitionGroup />
+            <ItemGroup />
+        </Project>
+        """)
+
+    dep_conf_props = textwrap.dedent("""\
+        <?xml version="1.0" encoding="utf-8"?>
+        <Project ToolsVersion="4.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+          <ImportGroup Label="PropertySheets">
+            {transitive_imports}
+          </ImportGroup>
+          <PropertyGroup Label="UserMacros" />
+          <PropertyGroup>
+            <conan_{name}_props_imported>True</conan_{name}_props_imported>
+          </PropertyGroup>
+          <PropertyGroup Label="ConanVariables">
+            <Conan{name}CompilerFlags>{compiler_flags}</Conan{name}CompilerFlags>
+            <Conan{name}LinkerFlags>{linker_flags}</Conan{name}LinkerFlags>
+            <Conan{name}PreprocessorDefinitions>{definitions}</Conan{name}PreprocessorDefinitions>
+            <Conan{name}IncludeDirectories>{include_dirs}</Conan{name}IncludeDirectories>
+            <Conan{name}ResourceDirectories>{res_dirs}</Conan{name}ResourceDirectories>
+            <Conan{name}LibraryDirectories>{lib_dirs}</Conan{name}LibraryDirectories>
+            <Conan{name}BinaryDirectories>{bin_dirs}</Conan{name}BinaryDirectories>
+            <Conan{name}Libraries>{libs}</Conan{name}Libraries>
+            <Conan{name}SystemDeps>{system_libs}</Conan{name}SystemDeps>
+          </PropertyGroup>
+          <PropertyGroup>
+            <LocalDebuggerEnvironment>PATH=%PATH%;$(Conan{name}BinaryDirectories)
+            </LocalDebuggerEnvironment>
+            <DebuggerFlavor>WindowsLocalDebugger</DebuggerFlavor>
+          </PropertyGroup>
+          <ItemDefinitionGroup>
+            <ClCompile>
+              <AdditionalIncludeDirectories>$(Conan{name}IncludeDirectories)
+              %(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
```

Comment:
I prefer not to break lines in this _template_. Are we sure they don't affect how VisualStudio read these lines? They are introducing several spaces (and a new line) between `$(Conan{name}IncludeDirectories)` and `%(AdditionalIncludeDirectories)`, can it be an issue?